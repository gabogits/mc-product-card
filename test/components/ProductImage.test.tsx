import React from 'react';

import renderer from 'react-test-renderer';
import { ProductCard } from '../../src/components/ProductCard';
import { product2 } from '../data/product';
import { ProductImage } from '../../src/components/ProductImage';

describe('ProductImage', () => {
  test('debe de mostrar el componente con la imagen personalizada', () => {
    const wrapper = renderer.create(<ProductImage img="./image.png" />);

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('debe mostar el componente conla imagen del pruducto', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
