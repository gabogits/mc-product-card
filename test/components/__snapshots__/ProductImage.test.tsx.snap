// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage debe de mostrar el componente con la imagen personalizada 1`] = `
<img
  alt="Product imge"
  className="productImg undefined undefined"
  src="./image.png"
/>
`;

exports[`ProductImage debe mostar el componente conla imagen del pruducto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product imge"
    className="productImg undefined undefined"
    src="./coffee-mug.png"
  />
</div>
`;
